src/Domain/Tags.hpp

```diff
@@ -40,19 +40,20 @@ namespace OptionTags {
 /// The input file tag for the DomainCreator to use
 template <size_t Dim, typename TargetFrame>
 struct DomainCreator {
-  using type = std::unique_ptr<::DomainCreator<Dim, TargetFrame>>;
+  using type = std::unique_ptr<domain::DomainCreator<Dim, TargetFrame>>;
   static constexpr OptionString help = {"The domain to create initially"};
 };
 }  // namespace OptionTags
 
+namespace domain {
 namespace Tags {
 /// \ingroup DataBoxTagsGroup
```

Comment:
I think with the `Tags` namespace now enclosed in `domain` you should document it as something like `\ref DataBoxGroup tags associated with the computational domain` to make it show up in Doxygen. Then you can also remove the `\ingroup DataBoxTagsGroup` from the tags below, since they are nicely listed in the namespace.